### DISCUSSION TITLE
Can we change date format and information order?

### DISCUSSION THREAD

**Discussion by rodhash** [rbong/vim-flog#150] [General]
Tried a few different ways two things: the date format to a shorter format, and the order of data ideally to: commit hash, author, msg, date.

Tried messing with these vars but seems to have no effect:
```vim
let g:flog_default_opts = {
            \ 'date': 'short',
            \ }

let g:flog_permanent_default_opts = {
            \ 'date': 'short',
            \ }

let g:flog_order_types = [
  \ { 'name': 'author', 'args': '--author-date-order' },
  \ { 'name': 'date', 'args': '--date-order' },
  \ { 'name': 'topo', 'args': '--topo-order' },
  \ ]
```

Suggestions?

**Comment by rbong** [comment]
**Setting the default date format:**

The date format can be specified by the "date" option.

It can be specified using `Flog -date=...`, the `g:flog_permanent_default_opts` setting (if you don't want the option to be cleared when resetting args), or `g:flog_default_opts` (if you're okay with it being cleared).  The way you're setting the default date format in your example is exactly correct.

This Flog argument corresponds to the `git log --date=...` argument - in other words setting `:Flog -date=my-date-format` is exactly the same as running `git log --date=my-date-format`.

You can view the documentation for this Git argument by running the command `man git log` and searching for `--date` (using the `/` key like in Vim) if you have Git manpages installed, or you can [read the documentation for `git log --date=...` online](https://git-scm.com/docs/git-log#Documentation/git-log.txt---dateltformatgt).

In general, when the documentation available in `:help flog` says something corresponds to a `git log` command argument or another subcommand, refer to the Git documentation for that command in this same way.

There are several special formats for `-date`/`--date`, including `short` like you've shown in your example.

**Custom date formats:**

Of particular interest to you may be the `--date=format:...` option, which is a special format that allows you to specify custom date formats. From the `git-log` manpage:

>--date=format:...  feeds the format ...  to your system strftime, except for %s, %z, and %Z, which are handled internally. Use --date=format:%c to show the date in your system locale’s preferred format. See the strftime manual for a complete list of format placeholders. When using -local, the correct syntax is --date=format-local:....

So for example in Flog, you can do something like `:Flog -date=format:%Y-%m-%d` to show the year, month, and day. For the full format specification, see `man strftime` or [an online copy of the strftime implementation for your OS manpage](https://www.man7.org/linux/man-pages/man3/strftime.3.html). The example I gave will work in most if not all Unix based systems.

**Changing the order of data:**

To change the order of data in the log, use the "format" option. This corresponds to the `--pretty=format:...` argument in `git- log`. This can be specified similar to the "date" option, and you can find the Git documentation for it in the same way.

For the format you've outlined - commit hash, author, msg, date - use the format option value `%h %an %s %ad`. (Note that if you specify this directly in a command, spaces need to be escaped with a backslash).

**The g:flog_order_types setting:**

This setting is used to change how order options are cycled through when cycling through them with `goo`.

**Example config:**

Putting it all together:

```vim
let g:flog_permanent_default_opts = {
  \ 'date': '%Y-%m-%d',
  \ 'format': '%h %an %s %ad',
  \ }
```

Hope that helps.
More information is in `:help flog` for each of the options and settings detailed here.

**Reply by rodhash** [reply]
Amazing, really appreciate all the details it worked great!

However I noticed what seems to be a bug, using `-date=relative` it breaks the HL groups when the date format changes and take more spacing:

<img width="652" alt="image" src="https://github.com/user-attachments/assets/1c939edf-ff39-46ae-9ba3-7937d762b2f7">

One more example:

<img width="665" alt="image" src="https://github.com/user-attachments/assets/f576fb3c-4340-49d5-a2e2-061fe8335d8a">

**Reply by rbong** [reply]
Should be fixed now on master.

**Reply by rodhash** [reply]
Hey .. thanks for the prompt fix but I don't see any commit on master

**Reply by rbong** [reply]
Apologies, should be pushed now

**Reply by rodhash** [reply]
> Apologies, should be pushed now

Thanks a lot, it's working great!
